### DISCUSSION TITLE
issue with PETSc when installing

### DISCUSSION THREAD

**Discussion by qlyan913** [firedrakeproject/firedrake#3433] [Firedrake support]
I got the following error messages when I install the firedrake with netgen flag:

"
....
Depending on your platform, PETSc may take an hour or more to build!
Traceback (most recent call last):
  File "/home/arnoldd/yan00082/myPython/firedrake-install", line 1867, in <module>
    install("petsc/")
  File "/home/arnoldd/yan00082/myPython/firedrake-install", line 1057, in install
    build_and_install_petsc()
  File "/home/arnoldd/yan00082/myPython/firedrake-install", line 1169, in build_and_install_petsc
    check_call([python, "./configure", "PETSC_DIR={}".format(petsc_dir), "PETSC_ARCH={}".format(petsc_arch)] + petsc_options)
  File "/home/arnoldd/yan00082/myPython/firedrake-install", line 670, in check_call
    log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
  File "/usr/local/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/local/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/arnoldd/yan00082/myPython/firedrake/bin/python', './configure', 'PETSC_DIR=/home/arnoldd/yan00082/myPython/firedrake/src/petsc', 'PETSC_ARCH=default', '--with-zlib', '--download-scalapack', '--download-suitesparse', '--do
wnload-hypre', '--download-hdf5', '--download-superlu_dist', '--download-mumps', '--download-metis', '--download-ptscotch', '--with-debugging=0', '--download-hwloc', '--download-netcdf', '--with-fortran-bindings=0', '--with-shared-libraries=1', '--download-chaco', '--d
ownload-cmake', '--download-pastix', '--download-mpich', '--download-bison', '--with-c2html=0', '--download-pnetcdf']' returned non-zero exit status 1.
"

Here is the install log: 
[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/14461285/firedrake-install.log)

and the configure.log
[configure.log](https://github.com/firedrakeproject/firedrake/files/14461286/configure.log)

Any help would be greatly appre

ciated. Thank you!

**Comment by ksagiyam** [comment]
```
*********************************************************************************************
           UNABLE to CONFIGURE with GIVEN OPTIONS (see configure.log for details):
---------------------------------------------------------------------------------------------
                          Error running make; make install on MPICH
*********************************************************************************************
```
Could you also post `/home/arnoldd/yan00082/firedrake/src/petsc/configure.log`?

**Reply by qlyan913** [reply]
Here is the configure.log
[configure.log](https://github.com/firedrakeproject/firedrake/files/14461276/configure.log)

**Reply by connorjward** [reply]
It appears like installation just crashed when using `gmake` to build MPICH. I notice that PETSc is trying to compile with a very large number of threads (59). I could be totally wrong but perhaps your machine is running out of memory. You can try limiting the number of threads used by setting either (I think)
```
export PETSC_CONFIGURE_OPTIONS=--with-make-np=8
```
or
```
export GNUMAKEFLAGS=-j8
```

You should also make sure that your system has been recently updated.
